clients/admin-ui/src/features/integrations/hooks/useIntegrationAuthorization.ts

```diff
@@ -0,0 +1,54 @@
+import { useAPIHelper } from "common/hooks";
+
+import { useLazyGetAuthorizationUrlQuery } from "~/features/datastore-connections/datastore-connection.slice";
+import {
+  ConnectionConfigurationResponse,
+  ConnectionSystemTypeMap,
+} from "~/types/api";
+
+interface UseIntegrationAuthorizationProps {
+  connection?: ConnectionConfigurationResponse;
+  connectionOption?: ConnectionSystemTypeMap;
+  testData: {
+    authorized?: boolean;
+  };
+}
+
+interface UseIntegrationAuthorizationResult {
+  handleAuthorize: () => Promise<void>;
+  needsAuthorization: boolean;
+}
+
+export const useIntegrationAuthorization = ({
+  connection,
+  connectionOption,
+  testData,
+}: UseIntegrationAuthorizationProps): UseIntegrationAuthorizationResult => {
+  const [getAuthorizationUrl] = useLazyGetAuthorizationUrlQuery();
+  const { handleError } = useAPIHelper();
+
+  const handleAuthorize = async () => {
+    if (!connection?.key) {
+      return;
+    }
```

Comment:
Consider adding some error handling here, even if it's a toast message to the user.